pkg/backend/httpstate/state.go

```diff
@@ -333,6 +334,30 @@ func convertStepEventMetadata(md engine.StepEventMetadata) apitype.StepEventMeta
 	for _, v := range md.Diffs {
 		diffs = append(diffs, string(v))
 	}
+	var richDiff map[string]apitype.DiffKind
+	if md.RichDiff != nil {
+		richDiff = make(map[string]apitype.DiffKind)
+		for k, v := range md.RichDiff {
+			var d apitype.DiffKind
+			switch v {
+			case plugin.DiffAdd:
+				d = apitype.DiffAdd
+			case plugin.DiffAddReplace:
+				d = apitype.DiffAddReplace
+			case plugin.DiffDelete:
+				d = apitype.DiffDelete
+			case plugin.DiffDeleteReplace:
+				d = apitype.DiffDeleteReplace
+			case plugin.DiffUpdate:
+				d = apitype.DiffUpdate
+			case plugin.DiffUpdateReplace:
+				d = apitype.DiffUpdateReplace
```

Comment:
feels weird to have what seems to be two entirely equivalent enums.  can we just have one and reference the same one from both places?

Reply:
They are equivalent, yes, but we've generally considered decoupling the API types from internal engine types to be a best practice. The API types in particular have specific marshaling requirements that need not be reflected in the internal types.